framework/src/org/apache/cordova/CordovaPlugin.java

```diff
@@ -359,4 +361,38 @@ public boolean onReceivedClientCertRequest(CordovaWebView view, ICordovaClientCe
      */
     public void onConfigurationChanged(Configuration newConfig) {
     }
+
+    /**
+     * Called by the Plugin Manager when we need to actually request permissions
+     *
+     * @return              Returns the permission that was stored in the plugin
+     */
+
+    public String[] getPermissionRequest() {
+        return permissions;
+    }
+
+    public boolean hasPermisssion() {
+        for(String p : permissions)
+        {
+            if(PackageManager.PERMISSION_DENIED == cordova.getActivity().checkSelfPermission(p))
```

Comment:
I understand the need for multiple permissions with 1 callbackContext. An array of permisisons would lead to a simpler design:

``public void requestPermissions(CordovaPlugin plugin, int requestCode, string[] permissions);```

However, this code will need to be changed to either have private knowledge of what permissions the geoloc plugin needs or ask it somehow:
https://github.com/apache/cordova-android/pull/220/files#diff-eb00315f6b4fb38aba0833096a86bb8aR190

I think then we should be able to remove the `CordovaPlugin` base class methods and helpers for permissions.